test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt

```diff
@@ -0,0 +1,156 @@
+package permissions.dispatcher.test
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.Process
+import android.provider.Settings
+import android.support.v4.app.ActivityCompat
+import android.support.v4.app.AppOpsManagerCompat
+import android.support.v4.content.PermissionChecker
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Matchers
+import org.mockito.Mockito
+import org.powermock.api.mockito.PowerMockito
+import org.powermock.core.classloader.annotations.PrepareForTest
+import org.powermock.modules.junit4.PowerMockRunner
+import permissions.dispatcher.test.ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcher.*
+import permissions.dispatcher.PermissionRequest
+
+@Suppress("IllegalIdentifier")
+@RunWith(PowerMockRunner::class)
+@PrepareForTest(ActivityCompat::class, PermissionChecker::class,
+        AppOpsManagerCompat::class, Process::class, Settings::class, Build.VERSION::class, Uri::class)
+class ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest {
+    private lateinit var activity: ActivityWithSystemAlertWindowAllAnnotations
+
+    companion object {
+        private var requestCode = 0
+
+        @BeforeClass
+        @JvmStatic
+        fun setUpForClass() {
+            // TODO Reflection on Kotlin top-level properties?
+            requestCode = 7
```

Comment:
I really don't like this code but I'm not sure how to get private top level properties using kotlin-reflection...

Reply:
Yeah, I had the same problem once, no idea.

Reply:
I asked the question to reflect channel in Kotlinlang

https://kotlinlang.slack.com/archives/C0AVAB92L/p1512281918000028

let's see

Reply:
TIL of the `#reflect` channel on kotlinlang's Slack. Nice!

Reply:
well, I think at this moment there is no way to solve this problem...

Reply:
Did you open an issue on YouTrack, by the way? If so, could you share the URL?

Reply:
Not yet, but I just notice I don't even know how to access a private top property. I read some article that says there is no way to access private top property. I couldn't find it at this moment.

...well, let me open the issue.

Reply:
https://youtrack.jetbrains.com/issue/KT-21707

Reply:
Sadly, 

```
there is no way to access private top property
```

this statement is correct... 😢 

https://youtrack.jetbrains.com/issue/KT-21707#comment=27-2595536